mne_bids/write.py

```diff
@@ -664,10 +664,12 @@ def _write_raw_fif(raw, bids_fname):
     n_rawfiles = len(raw.filenames)
     if n_rawfiles > 1:
         split_naming = 'bids'
-        raw.save(bids_fname, split_naming=split_naming, overwrite=True)
+        raw.save(bids_fname, split_naming=split_naming, overwrite=True,
+                 fmt=raw.orig_format)
     else:
         # This ensures that single FIF files do not have the split param
-        raw.save(bids_fname, split_naming='neuromag', overwrite=True)
+        raw.save(bids_fname, split_naming='neuromag', overwrite=True,
+                 fmt=raw.orig_format)
```

Comment:
do we have same issue with pybv / brainvision?

Reply:
good point, yes there *is* an issue, see these lines:

https://github.com/mne-tools/mne-bids/blob/429a6b9a7586e83967d13e2ba39c3bf41d79772d/mne_bids/write.py#L699-L705

We are forcing pybv to write with `resolution=1e-9` and in format `binary_float32` (default), we are *not* checking what the original data format was.

note that this only occurs when data is *converted* to BrainVision, because if BrainVision raw is passed, the raw is simply copied since https://github.com/mne-tools/mne-bids/pull/475/files

Reply:
> note that this only occurs when data is _converted_ to BrainVision,

So it's not a problem, I suppose?

Reply:
> So it's not a problem, I suppose?

unless we lose information because the original data was in higher resolution. But with `1e-9` and float32, we are at least having a relatively high resolution? I don't know.

Reply:
If you anonymize EEG, it will be converted to `pybv` ... there isn't an option to keep the original data format

Reply:
> unless we lose information because the original data was in higher resolution. But with `1e-9` and float32, we are at least having a relatively high resolution? I don't know.

`1e-9` volts is a resulotion that is, like, 1 to 2 orders of magnitude higher than what is required for meaningful brain signals, no? (I don't know about iEEG data, what kind of signals do these electrodes produce?)

For `fmt`, `pybv` only supports `binary_float32` and `binary_int16`, so we should be good here.

@agramfort's concern with FIFF writing was also not about **losing** data, but actually blowing up the required storage. So as long as nobody complains about huge BV files, I guess we're good? We should check the file size to be sure, though.

Reply:
> If you anonymize EEG, it will be converted with `pybv` ... there isn't an option to keep the original data format

Ok this, to me, is an argument that we _should_ try to recover and preserve the original resolution. Does MNE's BV Raw object expose this info?

Reply:
> If you anonymize EEG, it will be converted with pybv ... there isn't an option to keep the original data format

But I added functions for BrainVision to change the data in place, so the binary amplitude values are not changed.

Reply:
right, but what if your original data format was not Brainvision?

Reply:
> right, but what if your original data format was not Brainvision?

then conversion takes place, a warning is emitted, and the data is written with resolution `1e-9`, float32 format.

Reply:
> We should check the file size to be sure, though.

I did a couple of checks writing an EDF file to BV using different resolutions, and the resulting file size was always the same --> so I guess as long float32 is being used, the "resolution" parameter doesn't do a lot anyhow. With the defaults of float32, and resolution 1e-9 we are on the safe side I think.

Reply:
I'm currently looking into this a bit too